python/app/action_manager.py

```diff
@@ -0,0 +1,196 @@
+    UI_AREA_MAIN = 0x1
+    UI_AREA_DETAILS = 0x2
+    
+    def __init__(self, main_dialog):
+        """
+        Constructor
+        
+        :param main_dialog: Main sg panel dialog object
+        """        
+        self._app = sgtk.platform.current_bundle()
+        self._dialog = main_dialog
+    
+    def get_actions(self, sg_data, ui_area):
+        """
+        Returns a list of actions for an entity
+        
+        :param sg_data: Shotgun data for a publish
+        :param ui_area: Indicates which part of the UI the request is coming from. 
+                        Currently one of UI_AREA_MAIN, UI_AREA_DETAILS and UI_AREA_HISTORY
+        :returns: List of QAction objects, ready to be parented to some QT Widgetry.
+        """
+        if sg_data is None:
+            return []
+        
+        # check if we have logic configured to handle this
+        action_defs = []
+        all_mappings = self._app.get_setting("action_mappings")
+        if all_mappings.get(sg_data["type"]):
+            
+            mappings = all_mappings[ sg_data["type"] ]
+
+            # this is now a list of items, each a dictioary
+            # with keys filters and actions
+            # [{'filters': {}, 'actions': ['assign_task']}]
+
+            # now cull out actions that don't match our filters
+            actions_to_evaluate = []
+            for mapping in mappings:
+                actions_def = mapping["actions"]
+                filters_def = mapping["filters"]
+                if filters_def is None or len(filters_def) == 0:
+                    # no filters to consider 
+                    actions_to_evaluate.extend(actions_def)
+                else:
+                    # filters are on the form 
+                    # field_name: value
+                    for (field_name, field_value) in filters_def.iteritems():
+                        
+                        # resolve linked fields into a string value                        
+                        sg_value = sg_data.get(field_name)
+                        if isinstance(sg_value, dict):
+                            sg_value = sg_value.get("name")
+                        
+                        # check if the filter is valid
+                        if sg_value == field_value:
+                            actions_to_evaluate.extend(actions_def)
+                            
+            if len(actions_to_evaluate) > 0:
+                # no actions to run through the hook
+        
+                # cool so we have one or more actions
+                # call out to hook to give us the specifics.
+                
+                # resolve UI area
+                if ui_area == self.UI_AREA_DETAILS:
+                    ui_area_str = "details"
+                elif ui_area == self.UI_AREA_MAIN:
+                    ui_area_str = "main"
+                else:
+                    raise TankError("Unsupported UI_AREA. Contact support.")
```

Comment:
the reason is that the string is used when passed to the hook and the constant is used in the code internally. It seemed overkill to define some sort of concept of constants to pass from the main code into the hook - more pragmatic to just pass a string. This is easy for a TD to use. But in the code itself, a string felt a bit unelegant, so seemed better to use a constant there. But yes, i totally agree it's a bit funny. And i stole this bit of code from the loader so I didn't reflect much aobut it this time around :)